internal/resolution/variable_sources/entity_sources/catalogsource/registry_bundle_to_entity.go

```diff
@@ -0,0 +1,183 @@
+package catalogsource
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/operator-framework/operator-registry/alpha/property"
+	catalogsourceapi "github.com/operator-framework/operator-registry/pkg/api"
+	"k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+)
+
+type UpgradeEdge struct {
+	property.Channel
+	Replaces  string   `json:"replaces,omitempty"`
+	Skips     []string `json:"skips,omitempty"`
+	SkipRange string   `json:"skipRange,omitempty"`
+	Version   string   `json:"version,omitempty"`
+}
+
+type DefaultChannel struct {
+	DefaultChannel string `json:"defaultchannel"`
+}
```

Comment:
Having thought more about this, IMO, we should not generate new properties in the bundle entities for package and channel level info.

My thoughts are:
1.  a separate variable source for upgrade edge constraints
2. do we need/want to keep the concept of a default channel in OLMv1? Maybe it isnt important to answer that right now, but by the same token, do we need to implement default channel logic right now?

Reply:
Channels likely don't matter till we're dealing with upgrades/specific versions (resolver picks latest semver on an ordered list of channels to install). Since we install by packageName with the operator-controller, we'd need to rely on the catalogSource/FBC/user constraints to declare this order. FBC does support channel priorities for channel ordering, we don't necessarily need to use defaultChannel except for backward compatibility.

We can also follow a channel-free route if we use some other method for weighting versions/upgrade edges in the resolver, maybe using a different set of version constraints.

I can see the Upgrade Edge becoming its own variable rather than a property on the EntitySource. Whether it needs to be from a separate variableSource or should be part of the variables generated by the CatalogSource EntitySource depends on how we'd define the edge constraints.

Reply:
> IIRC in OLMv0, the purpose of the default channel is for dependency resolution

I'd argue that the default channel concept in v0 is used beyond just dependency resolution. Eg you can create a Subscription to a package without specifying a channel, and that installs the latest from the default channel. That to me is the concept being used for installation (and yes that itself is dependency resolution, but probably not in the sense you were talking about @joelanford, I.e resolves the dependencies of this operator I installed). 

That to me is a natural concept that should stay regardless. It gives package authors a way of publishing different versions in different channels, while keeping a sane default in place. I personally am definitely used to this (am sure others are) when we're installing binaries with rpm/brew etc. 

> IMO, we should not generate new properties in the bundle entities for package and channel level info

Huge +1. It's probably better to have `Bundle Entities` only contain bundle level information, and `Package Entities` have package level information from the get go, so that clients we're building out now (deppy) are built with the knowledge of that from scratch, since that's the direction we want to go (I.e separate out bundle and package level metadata, and only include bundle level info inside bundles). I.e this means clients (like deppy) will have to think in terms of getting `Bundle Entities` and `Package Entities` to resolve dependencies, and not just `Bundle Entities` (we're still thinking this way coz that's the way information is laid out currently in v0 bundles, but I'm guessing @joelanford is also trying to say let's break out this information now so that our underlying components are expecting scoped out information, and then we can work our way up to teach package authors to rethink how they package their operators)

Reply:
> let's break out this information now so that our underlying components are expecting scoped out information

@joelanford @anik120 @perdasilva 
it makes sense to have the edge properties be contained in their own variable rather than be mixed with the bundle variable. Are you suggesting that these be omitted from the bundles for now, and added later when we extend the resolver to support upgrades?